src/testflinger_device_connectors/fw_devices/tests/test_firmware_update.py

```diff
@@ -0,0 +1,123 @@
+"""Test detect_device in firmware_update.py"""
+
+
+import unittest
+import pytest
+from unittest.mock import patch
+from testflinger_device_connectors.fw_devices import *
+from testflinger_device_connectors.fw_devices.firmware_update import (
+    detect_device,
+)
+
+
+vendor_cmd = "sudo cat /sys/class/dmi/id/chassis_vendor"
+type_cmd = "sudo cat /sys/class/dmi/id/chassis_type"
+
+
+class TestFirmwareUpdate(unittest.TestCase):
+    def mock_run_cmd_supported(*args, **kwargs):
+        """
+        Mock run_cmd for a HP All In One device, which has a supported
+        Device class.
+        """
+        if args[1] == vendor_cmd:
+            return 0, "HP", ""
+        elif args[1] == type_cmd:
+            return 0, "13", ""
+
+    def mock_run_cmd_unsupported(*args, **kwargs):
+        """
+        Mock run_cmd for an Intel Desktop device (201808-26453), which doesn't
+        have a supported Device class.
+        """
+        if args[1] == vendor_cmd:
+            return 0, "Default string", ""
+        elif args[1] == type_cmd:
+            return 0, "3", ""
+
+    def mock_run_cmd_fail(*args, **kwargs):
+        """
+        Mock run_cmd for a Rigado Cascade 500 device (201810-26506), which doesn't
+        provide dmi data.
+        """
+        if args[1] == vendor_cmd:
+            return (
+                1,
+                "cat: /sys/class/dmi/id/chassis_vendor: No such file or directory",
+                "",
+            )
+        elif args[1] == type_cmd:
+            return (
+                1,
+                "cat: /sys/class/dmi/id/chassis_type: No such file or directory",
+                "",
+            )
+
+    def mock_run_cmd_nossh(*args, **kwargs):
+        """
+        Mock run_cmd to simulate an unreachable device.
+        """
+        if args[1] == vendor_cmd:
+            return (
+                255,
+                "",
+                "ssh: connect to host 10.102.161.93 port 22: No route to host",
```

Comment:
replace with some simpler fake IP that's easier to read for things like this - it's all mock data anyway